eZ/Publish/Core/Persistence/Cache/ContentHandler.php

```diff
@@ -408,22 +411,21 @@ public function deleteTranslationFromDraft($contentId, $versionNo, $languageCode
      * For use when generating cache, not on invalidation.
      *
      * @param \eZ\Publish\SPI\Persistence\Content\ContentInfo $contentInfo
-     * @param bool $withFields Set to true if item contains fields which should be expired on relation or type updates.
      * @param array $tags Optional, can be used to specify other tags.
      *
      * @return array
      */
-    private function getCacheTags(ContentInfo $contentInfo, $withFields = false, array $tags = [])
+    private function getCacheTags(ContentInfo $contentInfo, array $tags = [])
     {
         $tags[] = 'content-' . $contentInfo->id;
 
-        if ($withFields) {
-            $tags[] = 'content-fields-' . $contentInfo->id;
-            $tags[] = 'content-fields-type-' . $contentInfo->contentTypeId;
-        }
-
         if ($contentInfo->mainLocationId) {
-            $locations = $this->persistenceHandler->locationHandler()->loadLocationsByContent($contentInfo->id);
+            // As we don't have paging support on loadLocationsByContent(), slice it to avoid overloading tag system
+            $locations = array_slice(
```

Comment:
yes, it's actually a separate issue, so should not have dealt with it here.

Had a user on community slack getting into issues as they had several thousand locations. I never dreamt about anyone using locations that heavy, tag system is not meant for that amount of tags, so don't see any other way then cut it some way or the other to avoid it taking down the system. But it should at least have a warning when it happens, so I'll move this to own PR.